src/main/js/controls/Line/helpers/helpers.js

```diff
@@ -250,6 +250,77 @@ const draw = (scale, config, canvasSVG, dataTarget) => {
             .remove();
     }
 };
+
+/**
+ * Extracts regions inside value and prepare a list of value region
+ *
+ * @param {object} dataTarget - Data points object
+ * @param {Function} getXDataValues - checks if the value is valid and etract value for x-axis
+ *
+ * @returns {Array} value region subset
+ */
+const getValueRegionSubset = (dataTarget, getXDataValues) => {
+    const valueRegionSubset = [];
+    const valueRegions = [];
+    let valueRegion = {
+        color: undefined,
+        values: []
+    };
+    let previousColor;
+    dataTarget.values.forEach((value) => {
+        if (
+            !utils.isEmpty(value.region) &&
+            !utils.isEmpty(value.region.start) &&
+            !utils.isEmpty(value.region.end)
+        ) {
+            // If the color is different, then move it to new region set
+            if (previousColor !== value.region.color) {
+                if (valueRegion.values.length > 0) {
+                    valueRegionSubset.push(valueRegion);
+                }
+                previousColor = value.region.color;
+                valueRegion = {
+                    color: value.region.color,
+                    values: []
+                };
+            }
+            previousColor = value.region.color;
+            valueRegion.color = value.region.color;
+            valueRegion.values.push({
+                x: getXDataValues(value.x),
+                start: value.region.start,
+                end: value.region.end
+            });
+        } else if (valueRegion.values.length > 0) {
+            valueRegionSubset.push(valueRegion);
+            previousColor = undefined;
+            valueRegion = {
+                color: undefined,
+                values: []
+            };
+        }
+    });
+    if (valueRegion.values.length > 0) {
+        valueRegionSubset.push(valueRegion);
+    }
+
+    // Add start and end of a valueRegion to new valueRegion,
+    // This is to cover the start and end data value with region.
+    valueRegionSubset.forEach((region) => {
+        valueRegions.push(region);
+        if (region.values.length > 1) {
```

Comment:
can we have multiple values in one region? if not why will we have length more than 1?